11_notify/willshen8/pkg/puppy/rest.go

```diff
@@ -0,0 +1,144 @@
+package puppy
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	"github.com/go-chi/render"
+	"github.com/sirupsen/logrus"
+)
+
+const lostPuppyServer = "http://localhost:8888/api/lostpuppy/"
+
+// RestHandler provides a puppy storer to store the puppies
+type RestHandler struct {
+	store  Storer
+	client string
+}
+
+// NewRestHandler is a factory method that makes new storer
+func NewRestHandler(storer Storer) *RestHandler {
+	return &RestHandler{store: storer}
+}
+
+// HandleGet gets the puppy by id and displays the results.
+func (rh *RestHandler) HandleGet(w http.ResponseWriter, r *http.Request) {
+	id, err := strconv.Atoi(chi.URLParam(r, "id"))
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		http.Error(w, http.StatusText(http.StatusBadRequest)+": "+ErrInvalidInput.String(), http.StatusBadRequest)
+		return
+	}
+	puppy, readErr := rh.store.ReadPuppy(uint32(id))
```

Comment:
ok, I guess it's okay since you returned earlier, and that error doesn't doesn't get used anymore. In some places, I think it's easier to keep it distinct, so I don't get confused about all the types of error returned and don't get the error like such `pkg/puppy/rest.go:77:6: no new variables on left side of :=`